wagtail/wagtailcore/blocks/base.py

```diff
@@ -238,6 +238,39 @@ def get_context(self, value, context=None):
         })
         return context
 
+    def _get_context_with_context(self, value, context=None):
+        """
+        Temporary hack to accommodate Block subclasses created for Wagtail <1.6 which
+        have overridden `get_context` with the type signature:
+            def get_context(self, value)
+        and will therefore fail when passed a `context` kwarg.
+
+        In Wagtail 1.8, when support for context-less `get_context` methods is dropped,
+        this method will be deleted (and calls to it replaced with a direct call to `get_context`).
+        """
+        if accepts_context(self.get_context):
+            # this get_context method can receive a 'context' kwarg, so we're good
+            return self.get_context(value, context=context)
```

Comment:
I suggest we use the name `parent_context` for the new kwarg - it makes it a bit easier to understand and document what we mean by passing a context to `get_context`.

(Incidentally, `accepts_context` has been dropped from the codebase now, so I suggest you steal the `accepts_kwarg` implementation from https://github.com/wagtail/wagtail/pull/3188...)